src/aggregation_rules/aligned_time_rule.rs

```diff
@@ -96,4 +97,43 @@ mod tests {
         assert_eq!(c.open(), 13768.5);
         assert_eq!(c.close(), 13722.0);
     }
+
+    #[test]
+    fn aligned_time_rule_trigger_on_0() {
+        let trades: [Trade; 5] = [
+            Trade {
+                timestamp: 1712656800000,
+                price: 100.0,
+                size: 10.0,
+            },
+            Trade {
+                timestamp: 1712656815000,
+                price: 101.0,
+                size: -10.0,
+            },
+            Trade {
+                timestamp: 1712656860000,
+                price: 101.0,
+                size: -10.0,
+            },
```

Comment:
```suggestion
            Trade {
                timestamp: 1712656815000,
                price: 100.5,
                size: 10.0,
            },
            Trade {
                timestamp: 1712656860000,
                price: 101.0,
                size: -10.0,
            },
```
Would be good for both prices to be unique.